Supplementary material 3 for "A simple method to detect candidate overlapping genes in viruses using single genome sequences"
Timothy E. Schlub, Jan P. Buchmann, Edward C. Holmes

R code for the permutation and synonymous mutation procedure used here. “codon_permute_bootstrap” is an R function that takes a nucleotide sequence as input and returns the population of ORF lengths generated by bootstrapping in all reading frames. These populations  are returned . with names b.distrib0, com.b.distrib0, rev.b.distrib0, revcom.b.distrib0,b.distrib1, com.b.distrib1, rev.b.distrib1, revcom.b.distrib1,b.distrib2, com.b.distrib2, rev.b.distrib2, and revcom.b.distrib2, corresponding to reading frames +0, +c0, -0, -c0, +1, +c1, -1, -c1, +2, +c2 -2, and -c2, respectively. Each of these populations should subsequently should be unlisted (using the “unlist” inbuilt r function) and then converted to an empirical distribution using inbuilt r function “ecdf”, to enable easy calculation of P-values for different nucleotide lengths. The function “synonymous_mutation_bootstrap” is the equivalent function for the synonymous mutation test. These two functions rely on the custom functions “strReverse”, “complem”, “findORF”, and “synonymousMutaiton” to reverse strings, substitute for complementary nucleotides, find ORFs bookended by codons, and introduce random synonymous mutations respectively. 

The code also requires the r data file Codon_table.rds” to be imported as a dataframe titled "codon_table" before use.
